src/useWorker.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable no-console */
+import React from 'react'
+import createWorkerBlobUrl from './lib/createWorkerBlobUrl'
+import WORKER_STATUS from './lib/status'
+
+type Options = {
+  timeout?: number;
+  dependencies: string[];
+}
+
+const PROMISE_RESOLVE = 'resolve'
+const PROMISE_REJECT = 'reject'
+const DEFAULT_OPTIONS: Options = {
+  timeout: undefined,
+  dependencies: [],
+}
+
+/**
+ *
+ * @param {Function} fn the function to run with web worker
+ * @param {Object} options useWorker option params
+ */
+export const useWorker = <T extends (...fnArgs: any[]) => any>(fn: T, options: Options = DEFAULT_OPTIONS) => {
```

Comment:
`options` should be optional `options?: Options`, or isn't it necessary since I have a default?

Reply:
I did a test and define `options?: Options` as optional isn't necessary since we have `DEFAULT_OPTIONS`